### PR TITLE
VR-5723 Add --filename/-f to CLI's update endpoint

### PR DESCRIPTION
**Pull request by nhatsmrt**

### CODE REVIEW

File: client/verta/verta/_cli/deployment/update.py [hunk 1]
```diff
@@ -18,16 +18,22 @@ def update():
 @click.argument("path", nargs=1, required=True)
 @click.option("--run-id", "-r", help="Experiment Run to deploy. Cannot be used alongside --model-version-id.")
 @click.option("--model-version-id", "-m", help="Model Version to deploy. Cannot be used alongside --run-id.")
+@click.option("--filename", "-f", help="Path to JSON or YAML config file. Can only be used alongside --workspace.")
```
Comment (convoliution) [inline client/verta/verta/_cli/deployment/update.py]: Recommend also adding `type=click.Path()` with some appropriate options [[docs](https://click.palletsprojects.com/en/7.x/api/#click.Path)]!

File: client/verta/verta/_cli/deployment/update.py [hunk 2]
```diff
@@ -18,16 +18,22 @@ def update():
...
 @click.option("--workspace", "-w", help="Workspace to use.")
 # TODO: more options
-def update_endpoint(path, run_id, model_version_id, strategy, canary_rule, canary_interval, canary_step, env_vars, workspace):
+def update_endpoint(path, run_id, model_version_id, filename, strategy, canary_rule, canary_interval, canary_step, env_vars, workspace):
     """Update an endpoint.
     """
+    non_file_options = (run_id, model_version_id, strategy, canary_rule, canary_interval, canary_step, env_vars)
+
+    if filename and any(non_file_options):
+        raise click.BadParameter("--filename can only be used alongside --workspace.")
```
Comment (convoliution) [inline client/verta/verta/_cli/deployment/update.py]: This wording sounds like "`--filename` can only be used if `--workspace` is _also_ used.
Maybe something like "--filename can't be used alongside other options except for --workspace"?

**Review by convoliution — CHANGES_REQUESTED** [review]
Minor-ish suggestions

File: client/verta/verta/_cli/deployment/update.py (hunk 2 above)
Comment (nhatsmrt) [reply client/verta/verta/_cli/deployment/update.py]: Fixed.

File: client/verta/verta/_cli/deployment/update.py (hunk 1 above)
Comment (nhatsmrt) [reply client/verta/verta/_cli/deployment/update.py]: Fixed.

**Comment by nhatsmrt** [comment]
Suggestions incorporated, and conflicts resolved.

**Review by convoliution — APPROVED** [review]

**Comment by nhatsmrt** [comment]
Conflict resolved (again).

**Pull request merged**